pkg/kube/crds.go

```diff
@@ -318,3 +319,28 @@ func CleanCertmanagerResources(c kubernetes.Interface, ns string, config Ingress
 
 	return nil
 }
+
+func IsCertmanagerInstalled(c kubernetes.Interface, ns string) bool {
+	_, err := c.CoreV1().RESTClient().Get().RequestURI(fmt.Sprintf("/apis/certmanager.k8s.io/v1alpha1/namespaces/%s/issuers", ns)).Name(CertmanagerIssuerTesting).DoRaw()
+	if err == nil {
+		// existing test issuer found
+		_, err := c.CoreV1().RESTClient().Delete().RequestURI(fmt.Sprintf("/apis/certmanager.k8s.io/v1alpha1/namespaces/%s/issuers", ns)).Name(CertmanagerIssuerTesting).DoRaw()
+		if err != nil {
+			return false
+		}
+	}
+
+	// Try to create a cluster issuer just to test if cert-manager is installed in the cluster
+	json, err := yaml.YAMLToJSON([]byte(certmanager.Cert_manager_issuer_testing))
+	_, err = c.CoreV1().RESTClient().Post().RequestURI(fmt.Sprintf("/apis/certmanager.k8s.io/v1alpha1/namespaces/%s/issuers", ns)).Body(json).DoRaw()
+	if err != nil {
+		// Failed to create the test issuer, most likely no cert-manager is available
+		return false
+	}
+	_, err = c.CoreV1().RESTClient().Delete().RequestURI(fmt.Sprintf("/apis/certmanager.k8s.io/v1alpha1/namespaces/%s/issuers", ns)).Name(CertmanagerIssuerTesting).DoRaw()
+	if err != nil {
+		// cert-manager is there but somehow it failed to delete the fake issuer
+		return true
+	}
+	return true
```

Comment:
Just a thought: you can probably use the discovery API here instead of manually attempting to create/delete a (Cluster)Issuer resource. This should let you know if the `certmanager.k8s.io/v1alpha1` API group is available 😄

Reply:
@munnerz Thanks! This a good suggestion. I'll give it a try.